src/vsphere_cpi/lib/cloud/vsphere/nsxt_provider.rb

```diff
@@ -118,42 +124,58 @@ def initialize(client, default_vif_type)
     def add_vm_to_nsgroups(vm, ns_groups)
       return if ns_groups.nil? || ns_groups.empty?
       return if nsxt_nics(vm).empty?
-
-      logger.info("Adding vm '#{vm.cid}' to NSGroups: #{ns_groups}")
-      nsgroups = retrieve_nsgroups(ns_groups)
-
       lports = logical_ports(vm)
-      nsgroups.each do |nsgroup|
-        logger.info("Adding LogicalPorts: #{lports.map(&:id)} to NSGroup '#{nsgroup.id}'")
-        grouping_obj_svc.add_or_remove_ns_group_expression(
-          nsgroup.id,
-          *to_simple_expressions(lports),
-          'ADD_MEMBERS'
-        )
+      logger.info("Adding vm '#{vm.cid}' to NSGroups: #{ns_groups}")
+      Bosh::Retryable.new(
+          tries: 5,
```

Comment:
Why just 5 retries?

Reply:
`BOSH::Retryable` requires to specify the number of tries (default number of tries = 2)
I am trying to think of a way to model this to have infinite retries.
If not, can we use a high number - say 50 or 100? Would that be an enough number of retries?